BT mesh: Rename shadow variables

aescolar opened the pull request:
Rename shadow variables found by -Wshadow

After e38fc6de8ace95299595ff7963a02f9dfa712cd2
was merged, this cause CI failures.

----

Here a CI failure https://github.com/zephyrproject-rtos/zephyr/actions/runs/5937447483/job/16099892902?pr=60711#step:12:910

Fixes #61734

alxelax reviewed (dismissed):
It would be mature if the same problem is solved on the same manner. 
https://github.com/zephyrproject-rtos/zephyr/pull/61236/files#diff-9d35bfa2dc44279fbf0687382de4544e5ad427635016a5e5e80d77e546aa4fdfR34
It is better to rename global variable name here: https://github.com/zephyrproject-rtos/zephyr/pull/61733/files#diff-7584275c2da52cbcbe4efe8ca02fc9baacb02197f8db59677e8fc8828b0d1fbfL46
to `dh_pair`.
It should solve problem on the correct manner.

aescolar committed:
BT mesh: Rename shadow variables

Rename shadow variables found by -Wshadow

After e38fc6de8ace95299595ff7963a02f9dfa712cd2
was merged, this cause CI failures.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>
